### ISSUE TITLE
[Bug]: Invidious video proxing IP blocks

### ISSUE THREAD

**Issue by rjdg14** [FreeTubeApp/FreeTube#3892]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

1: Click on any video using either the Local or Invidious APIs
2: You will see the message "NotSupportedError: Failed to load because no supported source was found" as the video page loads
3: The video loading will infinitely hang and you will remain stuck on the thumbnail

### Expected Behavior

Clicking on most videos should result in normal playback after a short buffer period.

### Issue Labels

API issue, content not loading

### FreeTube Version

v0.18.0-nightly-3257 Beta

### Operating System Version

Windows 10 Pro 22H2 19045.3324

### Installation Method

.zip

### Primary API used

Local API

### Last Known Working FreeTube Version (If Any)

The previous Nightly build I tested (3220) initially worked fine but developed this issue. Upgrading to a newer nightly build did not fix the problem.

### Additional Information

The exact same problem appears to occur with both the Local and Invidious APIs, suggesting that this may be a bug in Freetube's code as opposed to an issue with either the YouTube or Invidious APIs. At the same time, the previous nightly build of Freetube (3220) that I downloaded worked fine until a couple of days ago.

### Nightly Build

- [X] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by absidue** [comment]
Disable the setting to proxy the video through Invidious, it's in the player settings.

**Comment by rjdg14** [comment]
> Disable the setting to proxy the video through Invidious, it's in the player settings.

That seems to have fixed it. Is this currently a bug/API issue that is yet to be fixed?

**Comment by absidue** [comment]
YouTube is IP blocking server IPs (Invidious, Piped and people running yt-dlp on servers are affected) on the video streaming endpoints.

Relevant Invidious issue: https://github.com/iv-org/invidious/issues/4045

**Comment by rjdg14** [comment]
> YouTube is IP blocking server IPs (Invidious, Piped and people running yt-dlp on servers are affected) on the video streaming endpoints.
> 
> Relevant Invidious issue: [iv-org/invidious#4045](https://github.com/iv-org/invidious/issues/4045)

It's most likely related to the recent legal action they threatened to take against Invidious (I read about this on Vice a few weeks back). I'd imagine that they have since realised that they are unable to take legal action against the project due to it being little more than a proxy, but they can probably find alternate means such as IP blocking to make it harder to use.

I tried loading various videos on an Invidious domain and they loaded fine on it.

Invidious was previously the main workaround for accessing age restricted YouTube videos without having to either verify your age in Europe/Australia (which I have already done on my Google account) or having to see a content warning every time you wanted to watch such a video (which was what put me off watching them on YouTube). In the past FreeTube would not load these videos via the Local API following changes Google made to their API in either 2020 or 2021, but now seems to be able to do so fine. It's possible that FreeTube is now taking the NewPipe approach and authenticating them through YouTube via an age verified Google account.

**Comment by absidue** [comment]
No the legal action is unrelated, if it were, they would be targeting just Invidious.
It's very easy for YouTube to detect and IP block servers that are using massive amounts of bandwidth on the video streaming endpoints.

Invidious doesn't proxy by default (neither does FreeTube) but if you enable proxying on either you get the issue, Piped on the other hand always proxies so they are completely unusable for streaming at the moment.

Neither FreeTube nor NewPipe nor Invidious use a Google account to bypass age restrictions, there is a workaround that works without needing an account, it just wasn't implemented in FreeTube before (in the nightlies since January, we are waiting on a release for an important dependency to fix some bugs before we do the 0.19.0 release candidate).

**Comment by absidue** [comment]
There is a big difference between, you watching a few videos on a VPN vs Invidious and Piped instances that use 100s of terabytes of bandwidth a month proxing video.

**Comment by thisguy786** [comment]
I am not sure this is an issue between YT and Invidious.

The Invidious videos play fine if you go to the same Invidious instance.  They only don't play in Freetube.

**Comment by absidue** [comment]
@thisguy786 FreeTube doesn't proxy the video through Invidious servers by default, neither does Invidious, you need to explicitly enable the setting in both for it to happen.

Some Invidious instances are frequently changing their IP addresses to try and avoid the YouTube's IP bans. The same instances that work with proxying enabled in your browser, will also work with proxying enabled in FreeTube.

**Comment by absidue** [comment]
@CrazzzzyChipmunk What is your preferred backend and have you enabled proxying videos through Invidious?

**Comment by absidue** [comment]
Does the syntax error disappear when you disable proxying?

**Comment by absidue** [comment]
I misread and thought the errors were in 0.19.0, glad to know they aren't and it really just is proxying that has a problem.

As for 0.18.0 feel free to use it if you really want to, but just know that it will just keep breaking more and more, with YouTube changes as it is already 10 months old. Please don't open any bug reports for any problems you face on 0.18.0 that don't exist on 0.19.0.

Ironically with your choice of using 0.18.0, only DASH gets proxied, audio only and legacy don't.

**Comment by sol8712** [comment]
Still getting this issue on most music related videos on the latest nightly 3505.

**Comment by absidue** [comment]
@sol8712 Switching the preferred API backend to the local API should fix that (as long as video proxying is disabled), according to issues on the Invidious GitHub repo, music videos are broken on Invidious.
e.g. https://github.com/iv-org/invidious/issues/4045#issuecomment-1755896459

**Comment by absidue** [comment]
@CrazzzzyChipmunk Which formats are you using in FreeTube (DASH, legacy or audio) and Invidious (DASH, HD720, Medium or Small)?

**Comment by absidue** [comment]
hd720 on Invidious is the equivalent of legacy 720p in FreeTube.

**Comment by absidue** [comment]
DASH gets proxied so with those setting it will get proxied. To use DASH 720p on the Invidious website you need to set it to DASH and then in their separate DASh quality drop down select 720p.

**Comment by absidue** [comment]
FreeTube only proxies when you have the setting enabled. When it is enabled both the local API and Invidious API streams will get proxied, when that setting is disabled, nothing is proxied, regardless of the selected API backend.

Newer versions of FreeTube can proxy all 3 formats: DASH, audio and legacy (just checked the code), 0.18.0 can only proxy DASh though.

The Invidious website cannot play DASH without proxying (browser limitation, CORS headers). FreeTube bypasses that limitation so it can play stuff without

**Comment by absidue** [comment]
I do know that some Invidious instances have had trouble in the past where they haven't returned proper proxied DASH URLs, so it is entirely possible that the only reason it worked on 0.28.0 is that FreeTube was trusting Invidious to return proxied URLs when we asked it to, but that they weren't. 0.19.0 stopped trusting Invidious and instead "force" proxied the URLs when Invidious failed to do it.

0.19.1 generates the DASH manifest inside FreeTube based on information in the Invidious API response, so that we can support multiple audio tracks when the Invidious API is selected, even though Invidious doesn't support it yet, but the information is there, so we can use it.

If you haven't tried the 0.19.1 version of FreeTube yet, I recommend you do. Proxying won't work properly because Invidious instances are either IP blocked and the ones that aren't, are getting throttled (they have a pull request open on their GitHub to fix the throttling but they haven't merged it yet) but the code on the FreeTube side is correct. So if you setup your own Invidious instance with their fix, proxying should work through your instance.

**Comment by treker7** [comment]
I think I'm seeing this same issue on the latest version of FreeTube - v0.19.1 Beta
This video will not load either through the Local API or through Invideus - https://youtu.be/c5NBimSuVOM
The error is - NotSupportedError: Failed to load because no supported source was found.
Full console logs attached.
[-1705789784295.log](https://github.com/FreeTubeApp/FreeTube/files/13999421/-1705789784295.log)
[-1705789725576.log](https://github.com/FreeTubeApp/FreeTube/files/13999423/-1705789725576.log)

**Comment by absidue** [comment]
@treker7 According to the logs, the calls to YouTube's servers timed out, so this is unrelated and either caused by something blocking the connection or you using some really slow VPN or proxy (e.g. tor).

**Comment by NathanMott** [comment]
I have been able to replicate this issue on v0.20.0 Beta with the below settings. If i switch Preferred API to local then it works fine. My question is why does the "Fallback to Non-Preferred Backend on Failure" not apply to this case?

Preferred API Backend: Invidious API
Fallback to Non-Preferred Backend on Failure: Enabled
invidious default instance: https://vid.puffyan.us
Default Video Format: DASH
Default Quality: 1080p
Allow DASH AV1 formats: enabled
Proxy Videos Through Invidious: Disabled

**Comment by mooreye** [comment]
I'm getting 

`NotSupportedError: Failed to load because no supported source was found.`

on some videos, without Invidious proxy enabled. It seems I can't play some specific videos at all.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Closing as we cant do anything about this.
